C7GameData/GameMap.cs

```diff
@@ -80,14 +85,31 @@ public bool isTileAt(int x, int y)
                 else
                     xInBounds = (x >= 1) && (x <= numTilesWide - 1);
             }
-            bool yInBounds = wrapVertically || ((y >= 0) && (y < numTilesTall));
-            return xInBounds && yInBounds && (evenRow ? (x%2 == 0) : (x%2 != 0));
+            return xInBounds && isRowAt(y) && (evenRow ? (x%2 == 0) : (x%2 != 0));
```

Comment:
Is the unusual braces syntax on lines 80-87 actually causing anything to happen, or is that just trying to say "this code is used to set the value of xInBounds"?  I saw that the other day and saw that you wrote it in November and was wondering what was up with it.  Usually if I wind up with braces in the middle of a function with no structure (if/for/try/etc.) around them, it's because I've made a mistake like accidentally deleting the line above them, so it made me wonder why those braces were there.

Reply:
You're right, the braces are only there to contain the code that initializes xInBounds. I got into the habit of doing that after dabbling in Rust a few years ago. Rust is an expression based language, meaning everything is an expression and hence evaluates to some value. Semicolons chain expressions together kind of like the comma operator in C/++. Blocks of code inside braces also evaluate to a value, the result of the last expression in the block, and so do if "statements" (there are no statements in Rust, only expressions). So in Rust you can initialize variables very neatly by assigning to them the value of a block. For example, that xInBounds initialization would look something like this (maybe not exactly right, it's been a few years):
```
let xInBounds = {
    if wrapHorizontally {
        true
    } else if evenRow {
        (x >= 0) && (x <= numTilesWide - 2)
    } else {
        (x >= 1) && (x <= numTilesWide - 1)
    }
};
```
Notice there's only the one assignment. Actually in this case you wouldn't even need the outer braces, you could just assign the if expression to xInBounds directly.

Anyway, I like this b/c it makes it clear that the purpose of the block of code is to initialize a variable and it also means any helper variables declared inside that block won't clutter up the outer scope. So I started emulating this in C/++, although of course there you need to do the variable assignment manually inside the block so it's not as neat. In C# it makes even less sense since C# doesn't have block scoping, so I've generally been avoiding doing this, for that reason and also because it's unusual.

Reply:
Ah yes, I also dabbled in Rust a few years ago.  I read about half of The Book, thought it was pretty neat, decided to port one of my old VB projects to Rust, realized I didn't understand the ownership concept as well as I thought when things weren't working, and a few days later Advent of Code started and I dove into that using C; so far I've yet to revisit Rust.  But I did like that sort of syntax, particularly when written like you did in your comment.

It doesn't feel very idiomatic in C#, though; a small helper method would probably be the usual way to do things?  As a one-off it's probably okay, but we probably generally want to stay idiomatic so newcomers who are familiar with C# recognize the constructs.